### PR TITLE
plan9 filesystem support

### PR DESCRIPTION
**Pull request by teknoraver**
used to share files between guest KVM and host

### CODE REVIEW

File: package/kernel/linux/modules/other.mk [hunk 1]
```diff
@@ -1055,3 +1055,39 @@ define KernelPackage/bmp085-spi/description
...
+define KernelPackage/virtio-pci
+  SUBMENU:=$(OTHER_MENU)
+  TITLE:=Virtio PCI support
+  KCONFIG:= CONFIG_VIRTIO CONFIG_VIRTIO_PCI
+  FILES:=\
+	$(LINUX_DIR)/drivers/virtio/virtio_pci.ko \
+	$(LINUX_DIR)/drivers/virtio/virtio.ko \
+	$(LINUX_DIR)/drivers/virtio/virtio_ring.ko
+  AUTOLOAD:=$(call AutoProbe,virtio virtio_ring virtio_pci)
+endef
```
Comment (blogic) [inline package/kernel/linux/modules/other.mk]: Auto probe should only list the last dependency, in this case virtio_pci. the rest will be probed.

File: package/kernel/linux/modules/other.mk [hunk 2]
```diff
@@ -1055,3 +1055,39 @@ define KernelPackage/bmp085-spi/description
...
+
+
+define KernelPackage/virtio-pci
+  SUBMENU:=$(OTHER_MENU)
+  TITLE:=Virtio PCI support
+  KCONFIG:= CONFIG_VIRTIO CONFIG_VIRTIO_PCI
+  FILES:=\
+	$(LINUX_DIR)/drivers/virtio/virtio_pci.ko \
+	$(LINUX_DIR)/drivers/virtio/virtio.ko \
+	$(LINUX_DIR)/drivers/virtio/virtio_ring.ko
```
Comment (blogic) [inline package/kernel/linux/modules/other.mk]: virtio and virtio_ring should be in their own kmod. this is duplicated in  the mmio module

File: package/kernel/linux/modules/other.mk [hunk 3]
```diff
@@ -1055,3 +1055,39 @@ define KernelPackage/bmp085-spi/description
...
+	$(LINUX_DIR)/drivers/virtio/virtio_mmio.ko
+  AUTOLOAD:=$(call AutoProbe,virtio virtio_ring virtio_mmio)
+endef
+define KernelPackage/virtio-mmio/description
+ This driver adds virtio MMIO support.
+endef
+
+$(eval $(call KernelPackage,virtio-mmio))
+
+
```
Comment (blogic) [inline package/kernel/linux/modules/other.mk]: 2 trailing blank lines

**Comment by blogic** [comment]
the first commit is missing the prefix in the subject and the SoB line

**Comment by nbd168** [comment]
What target(s) do you intend to use this with?

**Comment by teknoraver** [comment]
KVM virtual images, I share files from guest and host passing "-virtfs local,id=r,path=$HOME,security_model=none,mount_tag=home" to qemu cmdline

**Comment by nbd168** [comment]
You should not add kmod packages for virtio then, the kernel config already includes it.

I think it would be a good idea to add a virtio feature flag, which is auto-detected based on the kernel config. You could add that as a dependency for 9p packages.

File: package/kernel/linux/modules/other.mk [hunk 4]
```diff
@@ -1055,3 +1055,47 @@ define KernelPackage/bmp085-spi/description
...
 
 $(eval $(call KernelPackage,bmp085-spi))
+
+
+define KernelPackage/virtio
+  SUBMENU:=$(OTHER_MENU)
+  TITLE:=Virtio
+  KCONFIG:= CONFIG_VIRTIO
+  FILES:=$(LINUX_DIR)/drivers/virtio/virtio.ko
+  AUTOLOAD:=$(call AutoProbe,virtio)
```
Comment (blogic) [inline package/kernel/linux/modules/other.mk]: this line is not required

**Comment by nbd168** [comment]
Pulled into my staging tree, thanks.

**Pull request closed without merging**